zerver/tests/test_bots.py

```diff
@@ -1179,6 +1179,211 @@ def test_patch_bot_full_name_non_bot(self) -> None:
+        self.login("hamlet")
+        bot_info = {
+            "full_name": "The Bot of Hamlet",
+            "short_name": "hambot",
+        }
+        result = self.client_post("/json/bots", bot_info)
+        self.assert_json_success(result)
+        email = "hambot-bot@zulip.testserver"
+        bot = self.get_bot_user(email)
+
+        # Changing to the same short_name should be a no-op
+        bot_info = {
+            "short_name": "hambot",
+        }
+        result = self.client_patch(f"/json/bots/{bot.id}", bot_info)
+        self.assert_json_success(result)
+
+        bot.refresh_from_db()
+        self.assertEqual(bot.email, "hambot-bot@zulip.testserver")
+
+    def test_patch_bot_short_name_invalid_fake_email_domain(self) -> None:
+        self.login("hamlet")
+        # Create bot first without the invalid domain setting
+        bot_info = {
+            "full_name": "The Bot of Hamlet",
+            "short_name": "hambot",
+        }
+        result = self.client_post("/json/bots", bot_info)
+        self.assert_json_success(result)
+        email = "hambot-bot@zulip.testserver"
+        bot = self.get_bot_user(email)
+
+        # Now try to change short_name with invalid FAKE_EMAIL_DOMAIN
+        with override_settings(
+            FAKE_EMAIL_DOMAIN="invaliddomain", REALM_HOSTS={"zulip": "127.0.0.1"}
+        ):
+            bot_info = {
+                "short_name": "newbot",
+            }
+            result = self.client_patch(f"/json/bots/{bot.id}", bot_info)
+            error_message = (
+                "Can't change bot email until FAKE_EMAIL_DOMAIN is correctly configured.\n"
+                "Please contact your server administrator."
+            )
+            self.assert_json_error(result, error_message)
+
+    def test_patch_bot_short_name_unexpected_email_format(self) -> None:
+        """Test the fallback case when bot email doesn't end with '-bot'."""
+        self.login("hamlet")
+        bot_info = {
+            "full_name": "The Bot of Hamlet",
+            "short_name": "hambot",
+        }
+        result = self.client_post("/json/bots", bot_info)
+        self.assert_json_success(result)
+        email = "hambot-bot@zulip.testserver"
+        bot = self.get_bot_user(email)
+
+        # Manually set bot email to an unexpected format (without -bot suffix)
+        # This tests the fallback case in the code
+        bot.email = "unexpected@zulip.testserver"
+        bot.save(update_fields=["email"])
+
+        # Try to change short_name - should use fallback logic
+        bot_info = {
+            "short_name": "newbot",
+        }
+        result = self.client_patch(f"/json/bots/{bot.id}", bot_info)
+        self.assert_json_success(result)
+
+        bot.refresh_from_db()
+        self.assertEqual(bot.email, "newbot-bot@zulip.testserver")
+
+    def test_patch_bot_short_name_address_value_error(self) -> None:
+        """Test ValueError from Address constructor."""
```

Comment:
Do we check this for bot creation case? If not I think we can skip here as well.